completion.bash

```diff
@@ -19,7 +19,7 @@
 _charmcraft()
 {
     local cur prev words cword cmd cmds
-    cmds=(build version login logout whoami)
+    cmds=(build version login logout whoami names upload revisions status release)
```

Comment:
this is starting to get big enough that it feels like something we'd want to start automatically generating.
Manual is still fine for now, but it does mean having to keep 2 things up to date whenever adding an argument or optional parameter.